drivers/adc-dac/ad5592r/ad5592r.c

```diff
@@ -131,6 +132,34 @@ int32_t ad5592r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
 	return 0;
 }
 
+/**
+ * Read  ADC sequencer.
+ *
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_read_adc_sequence(struct ad5592r_dev *dev,
+				  uint16_t in_adc_seq,
+				  uint16_t *adc_values)
+{
+	int32_t ret;
+
+	if (!dev)
+		return FAILURE;
+
+	for (uint8_t i = 0; i <= dev->num_channels; i++) {
```

Comment:
the same would be here also similar to the i2c part. Create a sequence read no just multiple single reads.

Figure 44. page 33 of the datasheet.